src/hotspot/share/gc/g1/g1CollectedHeap.hpp

```diff
@@ -306,73 +306,18 @@ class G1CollectedHeap : public CollectedHeap {
 
   void trace_heap(GCWhen::Type when, const GCTracer* tracer) override;
 
-  // These are macros so that, if the assert fires, we get the correct
+  // These are assert functions so that, if the assert fires, we get the correct
   // line number, file, etc.
+public:
+  inline static void assert_heap_locked() NOT_DEBUG_RETURN;
+  inline static void assert_heap_locked_or_at_safepoint(bool should_be_vm_thread) NOT_DEBUG_RETURN;
+  inline static void assert_heap_locked_and_not_at_safepoint() NOT_DEBUG_RETURN;
+  inline static void assert_heap_not_locked() NOT_DEBUG_RETURN;
+  inline static void assert_heap_not_locked_and_not_at_safepoint() NOT_DEBUG_RETURN;
+  inline static void assert_at_safepoint_on_vm_thread() NOT_DEBUG_RETURN;
+  inline static void assert_used_and_recalculate_used_equal(G1CollectedHeap* g1h) NOT_DEBUG_RETURN;
```

Comment:
I think the intent was to leave these at namespace scope, rather than putting them in this class.
And perhaps move some of them to some more shared location, as many are pretty generic and
not specific to G1.